### PR TITLE
update build configurations to webgpu EP

### PR DESCRIPTION
**Pull request by fs-eire**
### Description

list out for tracking.

### CODE REVIEW

File: onnxruntime/core/providers/webgpu/program_manager.h [hunk 1]
```diff
@@ -30,7 +30,7 @@ class ProgramArtifact {
   const std::vector<int> shape_uniform_ranks;
 
   ProgramArtifact(ProgramArtifact&&) = default;
-  ProgramArtifact& operator=(ProgramArtifact&&) = default;
+  ProgramArtifact& operator=(ProgramArtifact&&) = delete;  // can't change const members.
```
Comment (snnn) [inline onnxruntime/core/providers/webgpu/program_manager.h:33]: Would it be safer if we disable both of them?

File: onnxruntime/core/providers/webgpu/webgpu_context.cc [hunk 2]
```diff
@@ -89,6 +92,8 @@ void WebGpuContext::Initialize(const WebGpuExecutionProviderInfo& webgpu_ep_info
   std::call_once(init_flag_, [this, &webgpu_ep_info]() {
```
[the remark this answers is not in the document]
Comment (fs-eire) [reply onnxruntime/core/providers/webgpu/webgpu_context.cc:92]: this is not a singleton pattern. `init_flag_` is a class member. I think static variable does not work in this case

File: onnxruntime/core/providers/webgpu/program_manager.h (hunk 1 above)
Comment (fs-eire) [reply onnxruntime/core/providers/webgpu/program_manager.h:33]: program artifact is used as a member of `ProgramManager` class:
```
std::unordered_map<std::string, ProgramArtifact> programs_;
```
and move constructor is needed

**Comment by fs-eire** [comment]
merged a few misc improvements and only list build configuration change.

**Review by snnn — APPROVED** [review]

**Review by snnn — APPROVED** [review]

**Comment by skottmckay** [comment]
Sorry - I didn't realize this branch was in a PR and I just merged in origin/main. 

@fs-eire do you want to update your branch with the latest origin/main to make the diffs reasonable again?

**Comment by fs-eire** [comment]
> Sorry - I didn't realize this branch was in a PR and I just merged in origin/main.
> 
> @fs-eire do you want to update your branch with the latest origin/main to make the diffs reasonable again?

I did it just now. But seems the status of PR is not updating to remove those unrelated changes.

**Comment by snnn** [comment]
You need to close this one and open a new PR.

**Comment by fs-eire** [comment]
> You need to close this one and open a new PR.

I tried to do that but in the change preview it still shows 140 files. Maybe need a git merge

**Comment by fs-eire** [comment]
> > You need to close this one and open a new PR.
> 
> I tried to do that but in the change preview it still shows 140 files. Maybe need a git merge

OK much better now.

File: onnxruntime/core/providers/webgpu/webgpu_execution_provider.cc [hunk 3]
```diff
@@ -435,7 +435,7 @@ std::unique_ptr<KernelRegistry> RegisterKernels() {
       KERNEL_CREATE_INFO(9, Cosh),
       KERNEL_CREATE_INFO(9, Asinh),
       KERNEL_CREATE_INFO(9, Acosh),
-      KERNEL_CREATE_INFO(9, Atanh),
+      // KERNEL_CREATE_INFO(9, Atanh),  TEMPORARY - Doesn't handle 1.0f -> inf with Metal
```
Comment (fs-eire) [inline onnxruntime/core/providers/webgpu/webgpu_execution_provider.cc:438]: how do we want to deal with Atanh on macOS/iOS? add a `#ifdef` for apple?

Comment (skottmckay) [reply onnxruntime/core/providers/webgpu/webgpu_execution_provider.cc:438]: If there any way to do it gracefully in the shader?

Comment (fs-eire) [reply onnxruntime/core/providers/webgpu/webgpu_execution_provider.cc:438]: what I can think is using `#ifdef APPLE` together with shader string generation.

for non-apple: just use builtin atanh
for apple: use custom_atanh:

fn custom_atanh(a) {
  if (a == 1.0f) {
       ...
  }
  return atanh(a);
}

Comment (skottmckay) [reply onnxruntime/core/providers/webgpu/webgpu_execution_provider.cc:438]: Not sure it's possible.

https://www.w3.org/TR/WGSL/#differences-from-ieee754
> A [shader-creation error](https://www.w3.org/TR/WGSL/#shader-creation-error) results if any [const-expression](https://www.w3.org/TR/WGSL/#const-expressions) of floating-point type overflows or evaluates to NaN or infinity.

We'd need to return +infinity, but it sounds like that's not possible.

**Pull request merged**